features/support/helpers.rb

```diff
@@ -16,6 +16,21 @@ def visit_renewal_details_page(reg_identifier)
   visit("#{Quke::Quke.config.custom['urls']['back_office_renewals']}/bo/renewing-registrations/#{reg_identifier}")
 end
 
+def sign_in_to_front_end_if_necessary(email)
+  @renewals_app = RenewalsApp.new
+
+  unless @renewals_app.waste_carriers_renewals_sign_in_page.displayed?
+    @renewals_app.waste_carriers_renewals_sign_in_page.load
+  end
+
+  return if page.has_text?("Signed in as")
```

Comment:
This looks like it returns if the user is signed in. But if we reuse this function, the user could be signed in as a different user, which would cause this step to fail.

Would this be safer?
`return if page.has_text?("Signed in as " + email)`

Reply:
In the case of the back office sign in helper we found on Friday that we were doing an extra unnecessary logout and login because of this too-specific check, even if already logged in with another user that had the same level of permission to perform the operation, hence, in this case, I decided to call the function with a more specific `_if_necessary` and leave the check generic.

Reply:
Merging :) If you think we should change this I am happy to open another PR :)

Reply:
OK - happy with that! I'll have another think when I come to reuse this function! :shipit: